File: app/src/main/java/com/semisonfire/cloudgallery/ui/main/MainPresenter.java
package com.semisonfire.cloudgallery.ui.main;

import com.semisonfire.cloudgallery.data.local.prefs.DiskPreferences;
import com.semisonfire.cloudgallery.ui.base.BasePresenter;

public class MainPresenter<V extends MainContract.View> extends BasePresenter<V>
        implements MainContract.Presenter<V>{

    public MainPresenter(DiskPreferences preferences) {
        super(preferences);
    }
}

File: app/src/main/java/com/semisonfire/cloudgallery/ui/main/MainPresenter.kt
package com.semisonfire.cloudgallery.ui.main

import com.semisonfire.cloudgallery.core.presentation.BasePresenter
import com.semisonfire.cloudgallery.data.remote.auth.Auth
import com.semisonfire.cloudgallery.data.remote.auth.AuthRepository
import io.reactivex.Observable

class MainPresenter(
  private val authRepository: AuthRepository
) : BasePresenter<MainContract.View>(), MainContract.Presenter {

  override fun saveToken(token: String) {
    authRepository.saveToken(token)
  }

  override fun getTokenListener(): Observable<Auth.AuthModel> {
    return authRepository.getAuthObservable().ofType(Auth.AuthModel::class.java)
  }
}
